src/liboslcomp/oslcomp.cpp

```diff
@@ -1149,8 +1158,30 @@ OSLCompilerImpl::check_write_legality (const Opcode &op, int opnum,
                sym->name().c_str(), opnum);
     }
 
-    // FIXME -- check for writing to globals.  But it's tricky, depends on
-    // what kind of shader we are.
+    // Check for writing to globals.
+    if (sym->symtype() == SymTypeGlobal) {
+        static ustring Ci("Ci"), P("P"), I("I"), N("N"), Ng("Ng");
+        static ustring dPdu("Pdu"), dPdv("Pdv"), u("u"), v("v");
+        static ustring time("time"), dtime("dtime"), dPdtime("dPdtime");
+        ShaderType stype = shadertype();
+        ustring symname = sym->name();
+        bool ok = true;
+        if (stype == ShaderType::Surface)
+            ok = (symname == Ci || symname == N);
```

Comment:
I don't think surface shaders are allowed to write to `N` ? (since most closure take in the bump normal)

Reply:
That's what the docs said.